mono/metadata/mono-config.c

```diff
@@ -618,7 +618,10 @@ mono_config_for_assembly_internal (MonoImage *assembly)
 	MONO_REQ_GC_UNSAFE_MODE;
 
 	MonoConfigParseState state = {NULL};
-	int got_it = 0, i;
+#ifndef TARGET_ANDROID
+	int got_it = 0;
+#endif
```

Comment:
we don't need to do C89-style variable declaration at the top so you can just move this down.